src/deepsparse/transformers/haystack/pipeline.py

```diff
@@ -0,0 +1,502 @@
+        pipeline_inputs = self.parse_inputs(*args, **kwargs)
+        if not isinstance(pipeline_inputs, self.input_schema):
+            raise RuntimeError(
+                f"Unable to parse {self.__class__} inputs into a "
+                f"{self.input_schema} object. Inputs parsed to {type(pipeline_inputs)}"
+            )
+
+        # run pipeline
+        if isinstance(pipeline_inputs.queries, List):
+            pipeline_results = [
+                self._haystack_pipeline.run(query=query, params=pipeline_inputs.params)
+                for query in pipeline_inputs.queries
+            ]
+        else:
+            pipeline_results = self._haystack_pipeline.run(
+                query=pipeline_inputs.queries, params=pipeline_inputs.params
+            )
+
+        outputs = self.process_pipeline_outputs(pipeline_results)
+
+        # validate outputs format
+        if not isinstance(outputs, self.output_schema):
+            raise ValueError(
+                f"Outputs of {self.__class__} must be instances of "
+                f"{self.output_schema} found output of type {type(pipeline_results)}"
+            )
+
+        return outputs
+
+    def process_pipeline_outputs(
+        self, results: Union[Dict[str, Any], List[Dict[str, Any]]]
+    ) -> BaseModel:
+        """
+        :results: list or instance of a dictionary containing outputs from
+            Haystack pipeline
+        :return: results cast to output_schema. If multiple inputs are passed,
+            then each field contains a list of values
+        """
+        if isinstance(results, List):
+            outputs = {key: [] for key in results[0].keys()}
+            for result in results:
+                for key, value in result.items():
+                    outputs[key].append(value)
+        else:
+            outputs = results
+
+        return self.output_schema(**outputs)
+
+    @property
+    def input_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return HaystackPipelineInput
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that outputs of this pipeline must comply to
+        """
+        return HaystackPipelineOutput
+
+    @property
+    def config_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that configs passed to this pipeline must
+            comply to
+        """
+        return HaystackPipelineConfig
+
+    def setup_onnx_file_path(self) -> str:
+        raise NotImplementedError()
+
+    def process_engine_outputs(
+        self,
+        engine_outputs: List[numpy.ndarray],
+        **kwargs,
+    ) -> BaseModel:
+        raise NotImplementedError()
+
+    def process_inputs(
+        self,
+        inputs: BaseModel,
+    ) -> Union[List[numpy.ndarray], Tuple[List[numpy.ndarray], Dict[str, Any]]]:
+        raise NotImplementedError()
```

Comment:
Same question about these not implemented. Maybe add a message into all these describing why they aren't implemented? And then as a side comment we may want to think about how to better architect these pipeline classes if we don't actually need to implement some of the required methods

Reply:
Haystack is a very non traditional pipeline because it requires inheritance from `Pipeline`, but does not do many of the things normal `Pipeline`s do